theme-kit/lib/createContentDeliveryClient.ts

```diff
@@ -0,0 +1,203 @@
+import type { Category, Culture, Newsroom, PrezlyClient } from '@prezly/sdk';
+import { ApiError, NewsroomGallery, SortOrder, Stories, Story } from '@prezly/sdk';
+
+const DEFAULT_PAGE_SIZE = 20;
+
+interface Params {
+    formats?: Story.FormatVersion[];
+    pinning?: boolean;
+    pageSize?: number;
+}
+
+export function createContentDeliveryClient(
+    prezly: PrezlyClient,
+    newsroomUuid: Newsroom['uuid'],
+    {
+        formats = [Story.FormatVersion.SLATEJS_V4],
+        pinning = false,
+        pageSize: defaultPageSize = DEFAULT_PAGE_SIZE,
+    }: Params = {},
+) {
+    const contentDeliveryClient = {
+        newsroom() {
+            return prezly.newsrooms.get(newsroomUuid);
+        },
+
+        languages() {
+            return prezly.newsroomLanguages.list(newsroomUuid).then((data) => data.languages);
+        },
+
+        async language(code?: Culture['code']) {
+            const languages = await contentDeliveryClient.languages();
+
+            return languages.find(
+                (lang) => (!code && lang.is_default) || lang.locale.code === code,
+            );
+        },
+
+        categories() {
+            return prezly.newsroomCategories.list(newsroomUuid, {
+                sortOrder: '+order',
+            });
+        },
+
+        async category(slug: Category['i18n'][string]['slug']) {
+            const categories = await contentDeliveryClient.categories();
+            return categories.find((category) =>
+                Object.values(category.i18n).some((t) => t.slug === slug),
+            );
+        },
+
+        featuredContacts() {
+            return prezly.newsroomContacts.search(newsroomUuid, {
+                query: {
+                    is_featured: true,
+                },
+            });
+        },
+
+        galleries(params: { page?: number; pageSize?: number; type?: `${NewsroomGallery.Type}` }) {
+            const { page, pageSize, type } = params;
+            const { limit, offset } = toPaginationParams({ page, pageSize });
+            return prezly.newsroomGalleries.search(newsroomUuid, {
+                limit,
+                offset,
+                scope: {
+                    status: NewsroomGallery.Status.PUBLIC,
+                    is_empty: false,
+                    type,
+                },
+            });
+        },
+
+        async gallery(uuid: NewsroomGallery['uuid']) {
+            try {
+                return await prezly.newsroomGalleries.get(newsroomUuid, uuid);
+            } catch (error) {
+                if (error instanceof ApiError && isNotAvailableError(error)) {
+                    return null;
+                }
+                throw error;
+            }
+        },
+
+        stories(params: {
+            search?: string;
+            category?: Pick<Category, 'id'>;
+            locale?: Pick<Culture, 'code'>;
+            page?: number;
+            pageSize?: number;
+            withFeaturedItems?: number;
```

Comment:
This parameter seems to be unused

Reply:
06f784a86e0d3104cf6c38e422ccee71be73b624